tools/src/tester/SupplementalChapterEvaluator.ts

```diff
@@ -0,0 +1,58 @@
+/*
+* Copyright OpenSearch Contributors
+* SPDX-License-Identifier: Apache-2.0
+*
+* The OpenSearch Contributors require contributions made to
+* this file be licensed under the Apache-2.0 license or a
+* compatible open source license.
+*/
+
+import ChapterReader from "./ChapterReader";
+import { StoryOutputs } from "./StoryOutputs";
+import { extract_output_values, overall_result } from "./helpers";
+import { ChapterEvaluation, Result } from "./types/eval.types";
+import { SupplementalChapter } from "./types/story.types";
+
+export default class SupplementalChapterEvaluator {
+    private readonly _chapter_reader: ChapterReader;
+
+    constructor(chapter_reader: ChapterReader) {
+        this._chapter_reader = chapter_reader;
+    }
+
+    async evaluate(chapter: SupplementalChapter, story_outputs: StoryOutputs): Promise<{ evaluation: ChapterEvaluation, evaluation_error: boolean }> {
+        const title = `${chapter.method} ${chapter.path}`
+        const response = await this._chapter_reader.read(chapter, story_outputs)
+        const status = chapter.status ?? [200, 201]
+        const output_values = extract_output_values(response)
```

Comment:
This should read `response.output_values`, it's very business-logic specific to the reponse.